src/ProtocolFees.sol

```diff
@@ -45,36 +46,39 @@ abstract contract ProtocolFees is IProtocolFees, Owner {
     }
 
     /// @notice Fetch the protocol fee for a given pool
-    /// @dev the success of this function is false if the call fails or the returned fees are invalid
-    /// @dev to prevent an invalid protocol fee controller from blocking pools from being initialized
-    /// the success of this function is NOT checked on initialize and if the call fails, the protocol fees are set to 0.
+    /// @dev Revert if call to protocolFeeController fails or if return value is not 32 bytes
+    /// However if the call to protocolFeeController succeed, it can still revert if the return value is too large
+    /// @return protocolFee The protocol fee for the pool
     function _fetchProtocolFee(PoolKey memory key) internal returns (uint24 protocolFee) {
         if (address(protocolFeeController) != address(0)) {
-            uint256 controllerGasLimit = block.gaslimit.calculatePortion(BLOCK_LIMIT_BPS);
-
-            // note that EIP-150 mandates that calls requesting more than 63/64ths of remaining gas
-            // will be allotted no more than this amount, so controllerGasLimit must be set with this
-            // in mind.
-            if (gasleft() < controllerGasLimit) revert ProtocolFeeCannotBeFetched();
-
             address targetProtocolFeeController = address(protocolFeeController);
             bytes memory data = abi.encodeCall(IProtocolFeeController.protocolFeeForPool, (key));
 
             bool success;
             uint256 returnData;
             assembly ("memory-safe") {
                 // only load the first 32 bytes of the return data to prevent gas griefing
-                success := call(controllerGasLimit, targetProtocolFeeController, 0, add(data, 0x20), mload(data), 0, 32)
+                success := call(gas(), targetProtocolFeeController, 0, add(data, 0x20), mload(data), 0, 32)
                 // if success is false this wont actually be returned, instead 0 will be returned
```

Comment:
need to edit this comments , will revert with "ProtocolFeeCannotBeFetched" , not 0